mono/mini/llvmonly-runtime.c

```diff
@@ -778,9 +778,14 @@ init_llvmonly_method (MonoAotModule *amodule, guint32 method_index, MonoClass *i
 	res = mono_aot_init_llvmonly_method (amodule, method_index, init_class, context, lookup_context, error);
 	if (!res || !is_ok (error)) {
 		MonoException *ex = mono_error_convert_to_exception (error);
-		/* Its okay to raise in llvmonly mode */
-		if (ex)
-			mono_llvm_throw_exception ((MonoObject*)ex);
+		if (ex) {
+			/* Its okay to raise in llvmonly mode */
+			if (mono_llvm_only) {
+				mono_llvm_throw_exception ((MonoObject*)ex);
+			} else {
+				mono_set_pending_exception (ex);
```

Comment:
how can this code be reached? shouldn't this method only be called if `mono_llvm_only` is set?

Reply:
This method is going to be pretty hot code for all LLVM-using JIT modes.

Reply:
Every single method will do a self-init through this function. If that init fails, we will throw here.

Reply:
ok. that was not obvious to me because of the filename (`llvmonly-runtime.c`) and function name (`init_llvmonly_method ()`), but it makes sense due to the nature of this change.  we should change that, but can be done later in order to avoid noise in this PR.

Reply:
Yes, these should be moved somewhere else later, they were only used by llvmonly previously.

Reply:
I renamed these functions. Let me know if you'd like for me to move them.